File: Downloads/PetyMate-main/backend/src/main/java/com/petymate/service/VetService.java
package com.petymate.service;

import com.petymate.dto.PagedResponse;
import com.petymate.dto.ProductDto;
import com.petymate.dto.VetDto;
import org.springframework.data.domain.Pageable;
import java.math.BigDecimal;

public interface VetService {
    PagedResponse<VetDto.VetResponse> getVets(String city, String state, String specialization,
            Boolean verifiedOnly, BigDecimal minRating, BigDecimal maxFee, String sortBy, Pageable pageable);
    VetDto.VetResponse getVetById(Long id);
    void addReview(Long vetId, String email, ProductDto.ReviewRequest request);
}

File: Downloads/PetyMate-main/frontend/src/services/vetService.js
import api from './api';

export const vetService = {
  getVets: (params) => api.get('/api/vets', { params }),
  getVetById: (id) => api.get(`/api/vets/${id}`),
  addReview: (id, data) => api.post(`/api/vets/${id}/review`, data),
};
